shell/golangci-lint.sh

```diff
@@ -17,8 +17,45 @@ fi
 # Enable only fast linters, and always use the correct config.
 args=("--config=${workspaceFolder}/scripts/golangci.yml" "$@" "--fast" "--allow-parallel-runners")
 
-# trade memory for CPU when running the linters
-export GOGC=20
+# Determine the version of go and golangci-lint to calculate compatibility.
+GO_MINOR_VERSION=$(go version | awk '{print $3}' | sed 's/go//' | cut -d'.' -f1,2)
+GOLANGCILINT_VERSION=$(asdf_devbase_run golangci-lint --version | awk '{print $4}')
+GO_MINOR_VERSION_INT=${GO_MINOR_VERSION//./}
+GOLANGCI_LINT_VERSION_INT=${GOLANGCILINT_VERSION//./}
+
+# Go 1.20 requires >= golangci-lint 1.52.0
```

Comment:
I guess this type of checking isn't as safe as I'd like it to be... since it'll break on a 2.0 version 😢 

I feel like I could just make a note of that here and remove it when that happens.

Reply:
Add a `$GO_MINOR_VERSION_INT[0] < 2` check or however it is you just check the major version?